src/libcsg/csgapplication.cc

```diff
@@ -368,7 +368,8 @@ namespace votca {
 
 
                 } else {
-                    master->Run();
+                    master->Start();
+                    master->WaitDone();
```

Comment:
@JensWehner This is where the problem is. Master is a Thread object from tools. The Run method is not meant to be called except internally by the the start method. Secondly because WaitDone was not called before the delete, the thread was being deleted before it had finished causing all sorts of errors.

Reply:
ahh I see, really good catch. 

Thanks